### PR TITLE
schedulers: make hot region scheduler respect schedule limit

### PR DESCRIPTION
**Pull request by disksing**
## What have you changed? (required)
Currently, the HotRegionScheduler adjusts schedule limit by itself. See https://github.com/pingcap/pd/blob/1e8e857d365585af6841a54473a5aca071ede509/server/schedulers/hot_region.go#L393-L405
A problem was found in the test: when there are many hot regions, when some tikv-servers are added, the limit will be updated to a large value, causing in a large number of operators in a short period of time, resulting in a QPS drop.

## What are the type of the changes (required)?

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (required)?

Unit test

### CODE REVIEW

**Comment by siddontang** [comment]
Cool

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -237,6 +237,10 @@ func (h *balanceHotRegionsScheduler) calcScore(items []*core.RegionStat, cluster
 }
 
 func (h *balanceHotRegionsScheduler) balanceByPeer(cluster schedule.Cluster, storesStat core.StoreHotRegionsStat) (*core.RegionInfo, *metapb.Peer, *metapb.Peer) {
+	if h.limiter.OperatorCount(schedule.OpRegion) >= cluster.GetRegionScheduleLimit() {
```
Comment (rleungx) [inline server/schedulers/hot_region.go]: Maybe put this into a function just like `allowBalancePeer` in `balanceAdjacentRegionScheduler`?

Comment (disksing) [reply server/schedulers/hot_region.go]: good idea.

**Comment by disksing** [comment]
PTAL @rleungx

**Pull request merged**